pkg/cloudprovider/rhsm/subscription_manager.go

```diff
@@ -168,41 +170,31 @@ func (d *defaultRedHatSubscriptionManager) refreshToken() error {
 }
 
 func (d *defaultRedHatSubscriptionManager) findSystemsProfile(name string) (string, error) {
-	req, err := http.NewRequest("GET", d.apiURL, nil)
+	systemsInfo, err := d.executeFindSystemsRequest(0)
 	if err != nil {
-		return "", fmt.Errorf("failed to create fetch systems request: %v", err)
+		return "", fmt.Errorf("failed to retrieve systems: %v", err)
 	}
 
-	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", d.credentials.AccessToken))
-
-	res, err := d.client.Do(req)
-	if err != nil {
-		return "", fmt.Errorf("failed executing fetch systems request: %v", err)
-	}
-	defer res.Body.Close()
-
-	data, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return "", fmt.Errorf("failed while reading response: %v", err)
-	}
-
-	if res.StatusCode != http.StatusOK {
-		if res.StatusCode == http.StatusUnauthorized {
-			return "", errUnauthenticatedRequest
+	for _, system := range systemsInfo.Body {
+		if system.Name == name {
+			return system.UUID, nil
 		}
-		return "", fmt.Errorf("error while exeucting request with status code: %v and message: %s", res.StatusCode, string(data))
 	}
 
-	var fetchedSystems = &systemsResponse{}
-	if err := json.Unmarshal(data, fetchedSystems); err != nil {
-		return "", fmt.Errorf("failed while unmarshalling data: %v", err)
-	}
+	var counter = 1
+	for counter <= (numberOfRequest(systemsInfo.Pagination.Count, d.requestsLimiter) - 1) {
```

Comment:
I'm not 100% a friend of this logic. While it does work, it feels complicated to calculate the number of pages instead of just querying the API and ending the loop when `numberOfFetchedSystems < limit`.

Ooooh is that `systemsInfo.Pagination.Count` thing the reason why we made one initial request outside of this loop? If so I am even more in favor of changing the loop logic, unless I am overlooking someting?